### DISCUSSION TITLE
Move Hub: Built-in button changes built-in led color.

### DISCUSSION THREAD

**Discussion by oneauraaa** [pybricks/support#1256] [Q&A] [answered]
I have a **Move Hub** and I want to change the color of the hub's built-in led everytime when I press built-in hub button Here is my **NOT WORKING** code.

```
from pybricks.hubs import MoveHub
from pybricks.parameters import Color, Button
from pybricks.tools import wait, StopWatch

hub = MoveHub()
value = 1

hub.system.set_stop_button(None)

watch = StopWatch()
while watch.time() < 5000:

    if hub.button.pressed():
        hub.light.on(Color.GREEN)
        value = (value % 3) + 1
    else:
        hub.light.on(Color.RED)
        print(value)  

hub.system.set_stop_button(Button.CENTER)

wait(5000)

```

Thanks.

**Comment by BertLindeman** [comment, marked as answer]
Hello,

Maybe some more info might help?

What firmware do you use?

**To reproduce**
You already decribed that.

**Expected behavior**
What did you expect to happen instead?

Next to these remarks I ran this code on my move hub:
```python
from pybricks.hubs import MoveHub
from pybricks.parameters import Color, Button
from pybricks.tools import wait, StopWatch
from pybricks import version
print(version)
wait(2000)

hub = MoveHub()
value = 1

hub.system.set_stop_button(None)
hub.system.set_stop_button(None)
hub.system.set_stop_button(None)
hub.system.set_stop_button(None)

watch = StopWatch()
while watch.time() < 5000:

    if hub.button.pressed():
        hub.light.on(Color.GREEN)
        value = (value % 3) + 1
        wait(100)
    else:
        hub.light.on(Color.RED)
        print(value)
        wait(100)  

hub.system.set_stop_button(Button.CENTER)

wait(5000)


```

~~Seemed to me that the more times I set the `hub.system.set_stop_button` the more times pressing the center button does work.
Huh??~~
I used beta.pybricks.com with firmware version: `('movehub', '3.3.0b6', 'v1.20.0-23-g6c633a8dd on 2023-06-02')`

Will update firmware and retry..

Bert
 
[EDIT}
Updated the movehub to `3.3.0b8` and updated the program of the OP above, as I did not see any keypress really change the colors.

This program works for me:
```python

from pybricks import version

from pybricks.hubs import MoveHub
from pybricks.parameters import Color, Button
from pybricks.tools import wait, StopWatch

print(version)
wait(500)

hub = MoveHub()

# skipped Blue to show "normal" state
colors = {1: Color.BLACK, 2: Color.NONE, 3: Color.BROWN, 4: Color.CYAN,
          5: Color.GRAY, 6: Color.GREEN, 7: Color.MAGENTA, 8: Color.ORANGE,
          9: Color.RED, 10: Color.VIOLET, 11: Color.WHITE}

num_colors = len(colors)
print("color has", num_colors, "colors")

hub.system.set_stop_button(None)
value = 0

print("Ready to press the button to change the hub light color")
for __ in range(num_colors):  # do once for each color

    while not hub.button.pressed():
        wait(100)  # wait a bit if no button pressed
    value += 1
    print(hub.button.pressed(), "go to color", value, colors[value])
    hub.light.on(colors[value])
    wait(500)

# Back to "normal"
hub.light.on(Color.BLUE)
print("Set Button.CENTER as STOP button again")
hub.system.set_stop_button(Button.CENTER)
wait(10000)

```

Does this help?

Bert

**Comment by oneauraaa** [comment]
Yes your code works! Now I actually want to run the motors with this line up (A - B - C - D - A and B - C and D). Can you help me? I just tried myself but I get errors.
`I just updated to 3.3.0b8 right now`

My code:
```
from pybricks import version

from pybricks.hubs import MoveHub
from pybricks.parameters import Color, Button
from pybricks.tools import wait, StopWatch

print(version)
wait(500)

hub = MoveHub()

# skipped Blue to show "normal" state
colors = {1: 'A', 2: 'B', 3: 'C', 4: 'D'}

motors = Motor(colors[value])          

num_colors = len(colors)
print("color has", num_colors, "colors")

hub.system.set_stop_button(None)
value = 0

print("Ready to press the button to change the hub light color")
for __ in range(num_colors):  # do once for each color

    while not hub.button.pressed():
        wait(100)  # wait a bit if no button pressed
    value += 1
    motors.run(500)
    #hub.light.on(colors[value])
    wait(500)

# Back to "normal"
print("Set Button.CENTER as STOP button again")
hub.system.set_stop_button(Button.CENTER)
wait(10000)
```

**Reply by BertLindeman** [reply]
This question is not really in line with the original question.

I will just give one hint here, but in case of a different question, a new discuss item might be more clear later.

The motors need a valid Port variable, like "Port.A" "Port.B" etc. Just "A" "B" are not valid Ports.

To start working with motors on the move hub look at the examples in [pybricks docs on motors](https://docs.pybricks.com/en/stable/pupdevices/motor.html)
And just for fun run the examples and try to understand.

Another python hint:
Keep your comments in line with the code. Makes it better to later understand what you meant.

Happy coding,

Bert

**Reply by oneauraaa** [reply]
It wouldnt be bad if I open a new thread? Because it would like duplicate thread...

**Reply by BertLindeman** [reply]
No, just to have clear issues that are later useful for others.
If we mix different questions, how would we later search for the best answers?

Bert

**Reply by oneauraaa** [reply]
Yeah, right. Thank you again.
